pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -165,3 +176,51 @@ func (a Adapter) generateAndGetContainerConfig(componentName string, comp versio
 	containerConfig := a.Client.GenerateContainerConfig(*comp.Image, comp.Command, comp.Args, envVars, containerLabels)
 	return containerConfig
 }
+
+func (a Adapter) generateAndGetHostConfig() (container.HostConfig, error) {
+	// Convert the port bindings from env.yaml and generate docker host config
+	portMap, err := getPortMap()
+	if err != nil {
+		return container.HostConfig{}, err
+	}
+	hostConfig := container.HostConfig{}
+	if len(portMap) > 0 {
+		hostConfig = a.Client.GenerateHostConfig(false, false, portMap)
+	}
+	return hostConfig, nil
+}
+
+func getPortMap() (nat.PortMap, error) {
+	// Convert the exposed and internal port pairs saved in env.yaml file to PortMap
+	dir, err := os.Getwd()
```

Comment:
why are we using os.Getwd()? Won't it be inaccurate when we support context flag?

Reply:
Good point! I think the same concern applies on https://github.com/openshift/odo/blob/5a26a56426b5d4e214c0817c2820350d9b862e0a/pkg/odo/cli/component/devfile.go#L117 and https://github.com/openshift/odo/blob/5a26a56426b5d4e214c0817c2820350d9b862e0a/pkg/odo/cli/component/devfile.go#L131
We need to change to use context to get the envinfo after context is supported in experimental mode.  I've added comments in the code, and also created an issue to track:https://github.com/openshift/odo/issues/2959

Reply:
Thanks for creating the issue :+1: